undeclared production Babel dependency in v0.25.0

The npm package includes a babel dependency that is undeclared in package.json

specifically, line 3 of /lib/index.js:

```js
var _interopRequireDefault = require("@babel/runtime/helpers/builtin/interopRequireDefault");
```

## Steps to reproduce

Just initialize a new npm repo and add `yup` as a dependency - it can't be imported in node (I'm running node v9.11.1 but I don't think that's related)

```
$ yarn init
$ yarn add yup
...
$ node
> require('yup')
Error: Cannot find module '@babel/runtime/helpers/builtin/interopRequireDefault'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:548:15)
    at Function.Module._load (internal/modules/cjs/loader.js:475:25)
    at Module.require (internal/modules/cjs/loader.js:598:17)
    at require (internal/modules/cjs/helpers.js:11:18)
```

This appears to be breaking type checking in Flow because it can't resolve the module, although my dev build does appear to work.

--- a/package.json
+++ b/package.json
@@ -74,6 +74,7 @@
     "sinon-chai": "^3.0.0"
   },
   "dependencies": {
+    "@babel/runtime": "^7.0.0-beta.47",
     "fn-name": "~2.0.1",
     "lodash": "^4.17.10",
     "property-expr": "^1.2.0",

